kore/src/Kore/Profiler/Data.hs

```diff
@@ -115,31 +178,65 @@ getTimePicos = timeSpecToPicos <$> getTime Monotonic
     timeSpecToPicos TimeSpec {sec, nsec} =
          ((toInteger sec * 1000 * 1000 * 1000) + toInteger nsec) * 1000
 
-{- Times an action.
+traceStderr
+    :: ProfileEvent -> IO ()
+traceStderr
+    ProfileEvent { endPico = Nothing, tags }
+  =
+    traceIO (intercalate "-" tags ++ " {")
+traceStderr
+    ProfileEvent { startPico, endPico = Just end, tags }
+  =
+    traceIO
+        (  "} "
+        ++ intercalate "-" tags
+        ++ " "
+        ++ show
+            ( fromRational ((end - startPico) % 1)
```

Comment:
Why not,
```suggestion
            ( fromInteger (end - startPico)
```

Although, it seems like we are converting to and then from picoseconds; would it make sense to store the raw `TimeSpec` instead?